### PR TITLE
boards/cpu: added SLWSTK6220A and EZR32WG support

### PR DESCRIPTION
**Pull request by haukepetersen**
EDIT
based ~~on #3525 and partly on #3265.~~

Adding support for the Silicon Labs Wireless Eval Kit SLWSTK6220A and it's CPU, the Silicon Labs EZR32 Wonder Gecko.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
Wow, is this board really so expensive or am I looking wrong ([offer](http://www.mouser.de/ProductDetail/Silicon-Labs/SLWSTK6220A/?qs=bQrJpfqjX%252bI7O5yZW5XIwA%3D%3D))?

**Comment by OlegHahm** [comment]
Nice one - can test tomorrow.

**Comment by OlegHahm** [comment]
> Wow, is this board really so expensive or am I looking wrong (offer)? 

Not if you are a RIOT developer. ;-) Besides, it has a display!11!elf

**Comment by haukepetersen** [comment]
Oh yeah - display driver is next :-) This port so far is not worth anything without being able to print the RIOT logo on the display :-)

**Comment by PeterKietzmann** [comment]
To avoid collisions regarding the display driver, we should arrange each other. @YRaoul once stated he wants to write a driver for the LimiFrog board #3020. Even if I never compared displays of this class, I think it could be beneficial to generate a display API then (cause we already have three boards available with a display on it: LimiFrog, stm32f4(29VZ)discovery and this one).

File: boards/slwstk6220a/board.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "cpu.h"
+#include "periph/gpio.h"
+
+// #include <stdio.h>
+// #include "periph/uart.h"
```
Comment (thomaseichinger) [inline boards/slwstk6220a/board.c]: Please remove.

File: boards/slwstk6220a/board.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    /* initialize the boards LEDs */
+    gpio_init(LED0_PIN, GPIO_DIR_OUT, GPIO_NOPULL);
+    gpio_init(LED1_PIN, GPIO_DIR_OUT, GPIO_NOPULL);
+
+    // uart_init(STDIO, 115200, NULL, NULL, NULL);
+    // uart_write_blocking(STDIO, 'A');
+    // uart_write_blocking(STDIO, 'B');
+    // uart_write_blocking(STDIO, 'C');
+    // uart_write_blocking(STDIO, '\n');
```
Comment (thomaseichinger) [inline boards/slwstk6220a/board.c]: Please remove.

File: cpu/ezr32wg/periph/cpuid.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+void cpuid_get(void *id)
+{
+    uint32_t source_address[] = { SAMD21_CPUID_WORD0,
+                                  SAMD21_CPUID_WORD1,
+                                  SAMD21_CPUID_WORD2,
+                                  SAMD21_CPUID_WORD3};
+    memcpy(id, (void*) source_address, CPUID_ID_LEN);
+}
```
Comment (thomaseichinger) [inline cpu/ezr32wg/periph/cpuid.c]: Is this file valid?

File: cpu/ezr32wg/periph/gpio.c [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+     * case output: no pull resistors available, use default drive strength
+     *  case input: use input without filter, set pull-up, pull-down or no-pull
+     *              as given */
+    mode = (dir | (pushpull & 0x3));
+    port->MODE[pin_pos >> 3] &= ~(0xf << ((pin_pos & 0x7) * 4));
+    port->MODE[pin_pos >> 3] |= (mode << ((pin_pos & 0x7) * 4));
+    // if (dir == GPIO_DIR_OUT) {
+    //     port->MODE[pin_pos >> 3] &= ~(0xf << ((pin_pos & 0x7) * 4));
+    //     port->MODE[pin_pos >> 3] |= (0x4 << ((pin_pos & 0x7) * 4));
+    // }
```
Comment (thomaseichinger) [inline cpu/ezr32wg/periph/gpio.c]: Please remove if not needed.

File: cpu/ezr32wg/periph/timer.c [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ * @file
+ * @brief       Low-level timer driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (thomaseichinger) [inline cpu/ezr32wg/periph/timer.c]: Should not be needed.

File: boards/slwstk6220a/include/periph_conf.h [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+extern "C" {
+#endif
+
+/**
+ * @brief   Clock configuration
+ * @{
+ */
+#define CLOCK_CORECLOCK     (14000000U)     /* run @ 48MHz */
+
+#define CLOCK_HFXO          (48000000U)     /* external HF crystal speed */
```
Comment (thomaseichinger) [inline boards/slwstk6220a/include/periph_conf.h]: Not used anywhere.

File: cpu/ezr32wg/include/cpu_conf.h [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+ */
+#define CPU_DEFAULT_IRQ_PRIO            (1U)
+#define CPU_IRQ_NUMOF                   (FPUEH_IRQn + 1)
+#define CPU_FLASH_BASE                  FLASH_BASE
+/** @} */
+
+/**
+ * @brief   CPUID_ID_LEN length of cpuid in bytes
+ */
+#define CPUID_ID_LEN                    (16U) /* 128 bits long, 16 bytes long */
```
Comment (thomaseichinger) [inline cpu/ezr32wg/include/cpu_conf.h]: Is this implemented yet?

**Comment by haukepetersen** [comment]
@PeterKietzmann I completely agree. I will try to get a better idea of the display first and get back to you guys on this.

@thomaseichinger thanks for the review, I actually forgot to adjust the CPUID driver as well as to add the actual clock initialization sequence... Will fix

**Comment by haukepetersen** [comment]
- added correct CPUID driver
- fixed comments

Will now look into the clock initialization

**Comment by haukepetersen** [comment]
now a proper clock initialization is also in place.

**Comment by OlegHahm** [comment]
Confirmed working, needs a rebase.

**Comment by haukepetersen** [comment]
Will rebase as soon as the dependencies are merged.

File: boards/slwstk6220a/Makefile.include [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# define the cpu used by WSTK6220
+export CPU = ezr32wg
+export CPU_MODEL = ezr32wg330f256r60
+
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(shell ls -1 /dev/tty.SLAB_USBtoUART* | head -n 1)
```
Comment (thomaseichinger) [inline boards/slwstk6220a/Makefile.include]: s/SLAB_USBtoUART/usbmodem/

**Comment by thomaseichinger** [comment]
Except of the comment above, flashing does not work for me. jlink output below.
```
### Flashing Target ###
HEXFILE found
SEGGER J-Link Commander V4.90b ('?' for help)
Compiled Aug 14 2014 10:26:36

Script file read successfully.
Can not connect to J-Link via USB.
Processing script file...

Unknown command. '?' for help.

Downloading file [/Users/thomas/code/RIOT/examples/hello-world/bin/slwstk6220a/hello-world.bin]...Unspecified error -1

Reset delay: 0 ms
Reset type UNKNOWN: ???



Script processing completed.
```

**Comment by OlegHahm** [comment]
IIRC JLink version > 5.0 is required.

File: dist/tools/jlink/jlink.sh [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+#
+# now comes the actual actions
+#
+do_flash() {
+    test_config
+    test_hexfile
+    # clear any existing contents in burn file
+    echo -n "" > ${BINDIR}/burn.seg
```
Comment (thomaseichinger) [inline dist/tools/jlink/jlink.sh]: Please use `/bin/echo` in jlink.sh. `sh`'s builtin echo behaves strange on OS X.
```sh
sh-3.2$ echo -n hello
-n hello
sh-3.2$ /bin/echo -n hello
hellosh-3.2$
```

**Comment by haukepetersen** [comment]
rebased as the xtimer was merged...

**Comment by haukepetersen** [comment]
moved to support the old UART driver, so not waiting on anything anymore. Ready for review and merge.

**Comment by haukepetersen** [comment]
rebased on #3847

**Comment by haukepetersen** [comment]
adapted UART to changes from #4114

**Comment by haukepetersen** [comment]
all dependencies are merged -> ready :-)

**Comment by haukepetersen** [comment]
those vendor headers used windows line endings... fixed.

**Comment by haukepetersen** [comment]
added license header and set doxygen excludes, this should make Travis happy.

**Comment by cgundogan** [comment]
I was able to flash and run some examples on this board. Looks good. ACK

**Comment by haukepetersen** [comment]
next fix for the vendor headers: added missing `extern C` declarations

**Comment by haukepetersen** [comment]
Travis is happy -> go go go

**Pull request merged**